src/build/__init__.py

```diff
@@ -156,14 +158,25 @@ def __init__(self, srcdir, python_executable=sys.executable):
 
         self._build_system = build_system
         self._backend = self._build_system['build-backend']
-
+        self.script_dir = os.path.dirname(python_executable) if scripts_dir is None else scripts_dir
```

Comment:
Defaulting to `os.path.dirname(python_executable)` is implementation specific, please remove it.

Call it `self.scripts_dir` to be more consistent. Here and in the other places.

Reply:
> Defaulting to `os.path.dirname(python_executable)` is implementation specific

Felt like the less bad of just leaving it empty, it's only used for non-isolated builds. Not?

Reply:
No, it's public API...

You could always
```
python -c 'import sysconfig; print(sysconfig.get_path("scripts"))'
```
which would be the correct solution.

Reply:
Are we adding the `scripts_dir` to the path when build is run without isolation, or why should it default to `scripts_dir`?